code/datums/components/storage/concrete/fish_case.dm

```diff
@@ -0,0 +1,19 @@
+/datum/component/storage/concrete/fish_case
+	max_items = 1
+	can_hold_description = "fish and aquarium equipment"
+
+/datum/component/storage/concrete/fish_case/can_be_inserted(obj/item/I, stop_messages, mob/M)
+	. = ..()
+	/// Activate deferred components if any.
+	SEND_SIGNAL(I,COMSIG_AQUARIUM_BEFORE_INSERT_CHECK)
+	if(I.GetComponent(/datum/component/aquarium_content))
```

Comment:
all these getcomponents are rough, and indicate that components aren't really being utilized like they should